### DISCUSSION TITLE
Easy way to modify/extend features like Visibility.ts in v8

### DISCUSSION THREAD

**Discussion by jpresr** [TanStack/table#4067] [Q&A]
Hi, I was wondering if there was an easy way to modify the built-in features like : [Visibility.ts](https://github.com/TanStack/table/blob/main/packages/table-core/src/features/Visibility.ts)

Our usage requires custom visibility settings, and before we were able to duplicate the plugins, but now we are not able to in v8. Whats the best approach for duplicating/extending built-in features in v8?

Thanks!

**Comment by KevinVandy** [comment]
Not sure if its an anti-pattern or not... but you can add your own state and setState functions to the table instance with some typescript trickery (and some ts-ignores). Or if you're just using JS, just follow this code pattern and ignore the TS.

You can see some custom state I've overridden the TS types with [here](https://github.com/KevinVandy/material-react-table/blob/87678781a1044ef6d0fa866c9851d4130b3f5dd9/src/MaterialReactTable.tsx#L133)

And then how I store a bunch of that state and the setStates right on the root of the instance [here](https://github.com/KevinVandy/material-react-table/blob/87678781a1044ef6d0fa866c9851d4130b3f5dd9/src/table/MRT_TableRoot.tsx#L260)

**Reply by jpresr** [reply]
There are some custom methods and callbacks we need to add that we unfortunately can't do with using state and setState. What we would want is to define our own TableFeature and be able to replace the existing features  [array](https://github.com/TanStack/table/blob/main/packages/table-core/src/core/instance.ts#L45) with our own features array.

**Reply by KevinVandy** [reply]
Guess I'm not sure why you can't just implement that functionality around the react-table instance with its own separate logic, and you don't have to put it on the instance at all.
Is this just wanting the old plugin system and not wanting to refactor that code?

**Reply by jpresr** [reply]
It's more about some of the existing features we add meta data and change some of the logic to. For example, we modify the sorting logic used [here](https://github.com/TanStack/table/blob/main/packages/table-core/src/features/Sorting.ts#L198). It seems like the easiest way to change it would be to replace the existing Sorting feature with our own copied and changed Sorting feature, which we could do with plugins. If I wanted to change the logic for that area, how would I do so in v8? I'm also not sure how you can set the correct order for features that are not built-in features. So if I were to make our own sorting wrapper over the instance, how would we know the order of filtering using the built-in feature would happen before the use of our wrapper over the instance? 

Thanks for the responses!
